solutions/fully-configurable/variables.tf

```diff
@@ -105,6 +105,42 @@ variable "allowed_network" {
   }
 }
 
+variable "secrets" {
+  type = list(object({
+    secret_group_name        = string
+    secret_group_description = optional(string)
+    create_access_group      = optional(bool, true)
+    access_group_name        = optional(string)
+    access_group_roles       = optional(list(string), ["SecretsReader"])
+    access_group_tags        = optional(list(string))
+  }))
+  description = "Secret Manager secret group and access group configurations. [Learn more](https://github.com/terraform-ibm-modules/terraform-ibm-secrets-manager/tree/main/solutions/fully-configurable/provisioning_secrets_groups.md)."
+  default = [
+    {
+      secret_group_name        = "General"
+      secret_group_description = "A general purpose secrets group with an associated access group"
+      create_access_group      = true
+      access_group_name        = "general-secrets-group-access-group"
+      access_group_roles       = ["SecretsReader"]
+    }
+  ]
+  validation {
+    error_message = "The name of the secret group cannot be null or empty string."
+    condition = length([
+      for group in var.secrets :
+      true if(group.secret_group_name == "" || group.secret_group_name == null)
+    ]) == 0
+  }
+  validation {
+    error_message = "When creating an access group, a list of roles must be specified."
+    condition = length([
+      for group in var.secrets :
+      true if(group.create_access_group && group.access_group_roles == null)
+    ]) == 0
+  }
+}
```

Comment:
I think this has to be set to `nullable = false` since there is a for loop used [here](https://github.com/terraform-ibm-modules/terraform-ibm-secrets-manager/blob/b548403a5f9bc2e2d95eb57e518d08312e53aed3/modules/secrets/main.tf#L22). You should add it to the secrets input in both the root module and submodule too.

Reply:
no, but the whole variable should be set to `nullable = false` because the for loop would fail if it was `null`. Use would just pass `[]` if they didnt wan't to create any groups